### DISCUSSION TITLE
Using an existing LB in Hetzner Cloud for control plane endpoint

### DISCUSSION THREAD

**Discussion by Stringls** [syself/cluster-api-provider-hetzner#734] [Q&A]
Hi! Is it possible to use an existing LB in Hetzner Cloud for Control plane? I didn't find any info about that in [docs/](https://github.com/syself/cluster-api-provider-hetzner/blob/main/docs/reference/hetzner-cluster.md#overview-of-hetznerclusterspec)
```
controlPlaneLoadBalancer.enabled | bool | true | no | Specifies if a load balancer should be created
```
If set `controlPlaneLoadBalancer.enabled` to false, then LB is not created, but I also cannot specify which one cluster should use

**Comment by janiskemper** [comment]
You have to specify the ControlPlaneEndpoint then!

**Reply by janiskemper** [reply]
Here it is in the docs
https://github.com/syself/cluster-api-provider-hetzner/blob/main/docs/reference/hetzner-cluster.md#usage-without-hcloud-load-balancer

**Reply by Stringls** [reply]
@janiskemper Sorry for late reply. So I can create a new LB11 in Hetzner Cloud (the same project where cluster is placed), add a new service `443 -> 6443`, specify as an endpoint the IPv4 of the LB and port 443, and Hetzner will add a Control plane machine as a target for this LB. Do I understand it correctly?

**Reply by batistein** [reply]
In general this is correct only that control-planes won't be added automatically as a target! For the caph controller you have disabled the management of a loadbalancer. So you need to manage the routing to the kubernetes api. Usually this feature is used if you have an own loadbalancer in front or you use a global loadbalancer. Also the controlPlaneEndpoint could be used to specify a DNS name. So you could also use your A-Records for some kind of loadbalancing...

**Reply by Stringls** [reply]
Cool, thank you for clarification @batistein @janiskemper
